modules/dbnd/src/dbnd/_core/task_run/task_run_sync_local.py

```diff
@@ -27,72 +33,140 @@ def __init__(self, task_run):
-                    self.outputs_to_sync.append((p_def, p_val, local_target))
+                self._sync_input_or_output(p_def, p_val, local_multitarget)
 
-                else:
-                    # Use DbndLocalFileMetadataRegistry to sync inputs only when necessary
-                    dbnd_meta_cache = DbndLocalFileMetadataRegistry.get_or_create(
-                        local_target
-                    )
+    def _local_cache_target(self, target_):
+        # type: (FileTarget) -> FileTarget
+        return target(
+            self.task_run.attemp_folder_local_cache,
+            os.path.basename(target_.path),
+            config=target_.config,
+        )
+
+    def _sync_input_or_output(self, param_definition, old_target, new_target):
+        # type: (ParameterDefinition, Type[DataTarget], Type[DataTarget]) -> None
+        if param_definition.kind == _ParameterKind.task_output:
+            # Output should be substituted for local path and synced post execution
+            self.outputs_to_sync.append((param_definition, old_target, new_target))
 
-                    # When syncing remote to local -> We can't use MD5, so use TTL instead
-                    if dbnd_meta_cache.expired or not local_target.exists():
-                        # If TTL is invalid, or local file doesn't exist -> Sync to local
-                        self.inputs_to_sync.append((p_def, p_val, local_target))
+        else:
+            self.inputs_to_sync.append((param_definition, old_target, new_target))
+            setattr(self.task, param_definition.name, new_target)
 
     def sync_pre_execute(self):
         if self.inputs_to_sync:
-            for p_def, p_val, local_target in self.inputs_to_sync:
+            for p_def, remote_target, local_target in self.inputs_to_sync:
                 # Input should be synced to local path and substituted
                 try:
-                    logger.info("Downloading  %s %s to %s", p_def, p_val, local_target)
-                    local_target.mkdir_parent()
-                    p_val.download(
-                        local_target.path,
-                        overwrite=local_target.config.overwrite_target,
+                    logger.info(
+                        "Downloading  %s %s to %s", p_def, remote_target, local_target,
                     )
-                except Exception as ex:
+                    if isinstance(remote_target, MultiTarget):
+                        for remote_subtarget, local_subtarget in zip(
+                            remote_target.targets, local_target.targets
+                        ):
+                            if (
+                                remote_subtarget.config.require_local_access
+                                and not remote_subtarget.fs.local
+                            ):
+                                # Use DbndLocalFileMetadataRegistry to sync inputs only when necessary
```

Comment:
all this code is just :  sync_remote_to_local(remote_target). (That can be used by Multi and regular). -- returns local target

Reply:
true, fixd